docs/contributing_sw/debugging_failed_builds.md

```diff
@@ -0,0 +1,180 @@
+```
+./eessi_container.sh --resume ${HOME}/pr370//EESSI-pilot-1698056784.tgz
+```
+
+For more info on using the EESSI container, see [here](../getting_access/eessi_container.md).
+
+### Start the Gentoo Prefix environment
+The next step is to start the Gentoo Prefix environment. 
+
+Before we start, check the current values of `${EESSI_CVMFS_REPO}` and `${EESSI_PILOT_VERSION}` so that you can reset them later:
+```
+echo ${EESSI_CVMFS_REPO}
+echo ${EESSI_PILOT_VERSION}
+```
+
+To do that, you need to run the `startprefix` command. However, we have several compatibility layers, and you'll need to run it for the one that matches the host node. For example, on an aarch64 (ARM) linux machine:
+```
+export EESSI_OS_TYPE=linux
+export EESSI_CPU_FAMILY=aarch64
+${EESSI_CVMFS_REPO}/versions/${EESSI_PILOT_VERSION}/compat/${EESSI_OS_TYPE}/${EESSI_CPU_FAMILY}/startprefix
+```
+
+if you are unsure, you can start the EESSI software environment (see next step) and check the values of `EESSI_OS_TYPE` and `EESSI_CPU_FAMILY` set by that initialization script. Note that you'll have to start over with a new shell (i.e. quit the container) and repeat the current step of starting the Gentoo Prefix environment, as the order of those two steps matters.
+
+Now, reset the `${EESSI_CVMFS_REPO}` and `${EESSI_PILOT_VERSION}` in your prefix environment
+```
+export EESSI_CVMFS_REPO=...
+export EESSI_PILOT_VERSION=...
+```
+
+!!! Note
+    By activating the Gentoo Prefix environment, the system tools (e.g. `ls`) you would normally use are now provided by Gentoo Prefix, instead of the container OS. E.g. running `which ls` after starting the prefix environment as above will return `/cvmfs/pilot.eessi-hpc.org/versions/2023.06/compat/linux/x86_64/bin/ls`. This makes the builds completely independent from the container OS.
+
+### Starting the EESSI software environment
+!!! Note
+    If you want to replicate a build with `generic` optimization (i.e. in `$EESSI_CVMFS_REPO/versions/${EESSI_PILOT_VERSION}/software/${EESSI_OS_TYPE}/${EESSI_CPU_FAMILY}/generic`) you will need to set `export EESSI_SOFTWARE_SUBDIR_OVERRIDE=${EESSI_CPU_FAMILY}/generic` before starting the EESSI environment.
+
+To activate the software environment, run
+```
+source ${EESSI_CVMFS_REPO}/versions/${EESSI_PILOT_VERSION}/init/bash
+```
+
+!!! Note
+    If you get an error `bash: /versions//init/bash: No such file or directory`, you forgot to reset the `${EESSI_CVFMS_REPO}` and `${EESSI_PILOT_VERSION}` environment variables at the end of the previous step.
+
+
+For more info on starting the EESSI software environment, see [here](../using_eessi/setting_up_environment.md)
+
+### Configure EasyBuild
+It is important that we configure EasyBuild in the same way as the bot uses it, with two small exceptions:
+
+- Our working directory will be different
+- Our installpath will be different
```

Comment:
Unless one wants to build into `/cvmfs/...`

Reply:
Agreed. I've had one issue that didn't show up when building somewhere else. But: I don't know what needs to change exactly to build in `/cvmfs/`. I guess we need to pass `--access rw` then? Does it then mount the writeable overlay? Or is there more that needs to be done?

As I understand, it is a bit slower to compile in the writeable overlay. What would be your idea: should we advise that by default, simply to stay as close to the bot as possible? Or not, because it is slower?

Reply:
Also, if we install there, is that saved into the tarball as well by `--save` and then restored by `--restore`? I.e. if I need to build some deps first,, then save, then restore later, will the deps be already there?

Reply:
Ok, `--access rw` and _not_ setting `EASYBUILD_INSTALLPATH` ourselves (i.e. relying on the one set by `configure_easybuild` works. This creates builds that actually get installed into the writeable overlay. That has the advantage of being even more similar to what the bot does. Hence, I was able to replicate also [this issue](https://github.com/EESSI/software-layer/pull/332#issuecomment-1775374260).

I think that is an improvement. I still add to the instructions the suggestion of setting an `EASYBUILD_INSTALLPATH`, but only _after_ having replicated the issue - as a way of identifying if the writeable overlay is somehow causing an issue.